### PR TITLE
purchase5assigniment

### PR DESCRIPTION
**Pull request by PrinceItam**
Made a new purchase with automation web application. help review please

### CODE REVIEW

**Review by bidemi20002000 — APPROVED** [review]

**Comment by bidemi20002000** [comment]
Hello Prince, your Test did not inherit the common steps. Our framework is design such that every test should inherit the common steps.
secondly, your assertion appears to be pointing to home page rather than confirm that the order was successful

**Review by lanrecoal2020 — CHANGES_REQUESTED** [review]
Your test is not pointed to test base at all. If there is a test base configured for the project, then all tests within the project should be pointing to testbase. See the commontest,java file to see how you use testbase and config properties variables to navigate to homepage.

**Comment by lanrecoal2020** [comment]
@PrinceItam NB.When you point ur test to testbase, you dont need to declare public Webdrive driver in your step definition file. see commonTest.java and see how you can us the @before annotation to innitialize the webdriver that is declared in testbase

**Comment by PrinceItam** [comment]
Changes made please review

File: src/test/java/Step_Definition/purchase5.java [hunk 1]
```diff
@@ -38,6 +42,7 @@ public void iPlaceAnOrder() throws InterruptedException {
         size.selectByVisibleText("M");
         page.selectgreen();
         driver.findElement(By.id("quantity_wanted")).clear();
+        driver.findElement(By.id("quantity_wanted")).sendKeys("65");
```
Comment (lanrecoal2020) [inline src/test/java/Step_Definition/purchase5.java]: please change lines 44 and 45 to page object format

File: src/test/java/Step_Definition/purchase5.java [hunk 2]
```diff
@@ -52,12 +57,14 @@ public void iPlaceAnOrder() throws InterruptedException {
         page.selectpayviacheck();
         page.clickconfirmorder();
         page.clickbackOrder();
+
     }
 
     @Then("^I should see my order history and reference$")
     public void iShouldSeeMyOrderHistoryAndReference() {
         String ExpectedMessage = "ORDER HISTORY";
```
Comment (lanrecoal2020) [inline src/test/java/Step_Definition/purchase5.java]: Please assert your successful order with the success message. This order history doesnt really confirm the order that the last test run made...

**Review by lanrecoal2020 — CHANGES_REQUESTED** [review]

**Review by lanrecoal2020 — APPROVED** [review]

**Pull request merged**